README.md

```diff
@@ -431,10 +435,15 @@ Example (find this in the test configuration):
 }
 ...
 ```
+When `driverPerSuite` is `true` the global `beforeEach` hook will have a `nemo` instance injected, but not when `driverPerSuite` is `false`
 
-Please note: When using the `driverPerTest` option, there will be no `nemo` instance in the `before`/`after` lifecycle
+Please note: When using the `driverPerTest` option, there will be no reliable `nemo` instance in the `before`/`after` lifecycle
 context.
 
+### InstanceId
+
+Each nemo instance will be assigned an `instanceId` property, that you can use to uniquely identify a nemo instance.
```

Comment:
We already have a `uid` tag to identify Nemo instances. Can you explain why a new unique ID is needed here?

Reply:
I must have missed it.  Which property is that?

Reply:
never mind, you said `uid`
I'll remove the new one.

Reply:
@grawk OK, I tried to replace `instanceId` with `uid` in https://github.com/krakenjs/nemo/blob/7fa2fe22453630d1d85098abbdf3a902b6e7cad8/lib/runner/mocha.js#L54 and got undefined.  Where and when exactly is this `uid` defined?  I think it might only be available for events, not nemo object itself.

Reply:
@grawk I found the value in `instanceConfig.profile.tags.uid` in `runner/mocha.js` but it's not unique per actual nemo-core instance injected into mocha hook `this` context.  Seems to be more of a "Nemo Runner profile / parallel" scope lifetime, which can spawn multiple nemo-core instances.

Reply:
Thanks @Noyabronok .. I had to go familiarize myself with that code again. Can you please let me know what is your use case for this additional property on the nemo object?

Reply:
@grawk Well, I initially added it to easily understand and confirm the lifetime of the `this.nemo` object.  So, right now it's just referenced in logging.  Maybe there are some other use cases, not sure.  I thought it was nice to keep since I already added it.

While we're at it, I'd like to confirm my understanding of the `tags.uid` property.  Readme.md refers to it as an instance for a process.  Seems like the purpose is to identify different process instances running in parallel. To me, that's not a Nemo instance, but more of a process instance, since each process can spawn many nemo-core instances, as I mentioned above.  

But, I guess it's a matter of language.  I see that Readme.md refers to each nemo-core instance as **webdriver**, as mentioned in the lifecycle section explaining `driverPerTest`.  So maybe, to keep things consistent with existing language, I should rename `this.nemo.instanceId` to `this.nemo.webdriverId` or `this.nemo.driverId`?

Reply:
You're right. The term "nemo instance" is unfortunately a bit overridden by those two meanings. You could consider the current `tags.uid` as representing a Mocha `Runner` instance. And within that instance there could be multiple `nemo` instances.

My trepidation to adding `nemo.instanceId` is that it adds a new property on the `nemo` object, which is currently fairly sparse. Do you think it would be ok to take out the new property from this PR and discussing it in an issue or future PR?

That would clear the way to merge/publish the next version which takes advantage of the newly fixed Mocha `start` event.

Reply:
@grawk I removed `instanceId` from this PR.  Please merge.  Thanks!